src/kernel.cr

```diff
@@ -125,22 +125,47 @@ end
 module AtExitHandlers
   @@running = false
 
+  private class_getter(handlers) { [] of Int32 -> }
+  private class_getter(exceptions) { [] of Exception }
```

Comment:
When would you have more than one?

Reply:
Currently never, I was just thinking that maybe some shards might want to add multiple exceptions.. But now that I rethink about it it doesn't make sense, or at least I shoukd apply YAGNI until someone needs this.